### PR TITLE
Avatar system

### LINKED ISSUE #39: Avatar system
**Specifications**
The proposal indicates a live avatar system.

We will need to determine the possibility of this.

Worse case we simply allow users to upload fixed images.

### PR DESCRIPTION
**Pull request by JZ4012**
### Team Monkey pull request checklist (Jungle edition)

-   [x] Remember to format your code! (`npm run prettier && npm run eslint`)
-   [x] Remember to run the tests! (`npm run test`)
-   [x] Don't forget comments and docstrings!
-   [x] Is it a breaking change? Remember to mention that!

### Description

Implements:
- user profile picture uploading and viewing functionality
- document upload and document download functionality.

Closes #39, #41, #42  

🐵🐒🙊🙉🙈

### CODE REVIEW

**Review by JZ4012 — COMMENTED** [review]
very good code +rep

File: src/client/filehandler/FileContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+export const FileContainer: React.FunctionComponent<Props> = (props: Props) => {
+    const [fileData, getFiles] = useDynamicFetch<
+        { [key: string]: FileStorageType },
+        { sid: string }
+    >("/filehandler/getFiles", { sid: props.sessionID }, true);
+
+    // Updates the list of files displayed.
+    useSocket("newfile", undefined, undefined, () => {
```
Comment (aXises) [inline src/client/filehandler/FileContainer.tsx:19]: Put this in a enum

File: src/server/server.ts [hunk 2]
```diff
@@ -63,6 +66,27 @@ app.use((req: Request, res: Response, next: NextFunction) => {
     next();
 });
 
+app.get(
```
Comment (aXises) [inline src/server/server.ts:69]: Can remove this

File: src/server/routes/filehandler.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+export const router = express.Router();
+
+router.get(
+    "/file/:sessionId/:fileId",
+    asyncHandler<{}, { sessionId: string; fileId: string }>(
+        async (req, res) => {
+            const session = await Session.findById(req.params.sessionId);
+            const file = session?.files?.get(req.params.fileId);
+            const contents: Binary = file?.file as any;
```
Comment (aXises) [inline src/server/routes/filehandler.ts:18]: Your `Session.files` type might be wrong, it's cast to any and assigned to `Binary` here when your type is `Buffer`.

File: src/server/routes/filehandler.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+            res.set(
+                "Content-disposition",
+                `attachment; filename=${file?.filename}`
+            );
+            res.set("Content-Type", file?.fileExtension);
+            res.end(contents.buffer);
```
Comment (aXises) [inline src/server/routes/filehandler.ts:25]: The error from deepscan here is caused by `file?.file`, remove the `?` and the cast which has hidden some of the problems.

**Review by aXises — CHANGES_REQUESTED** [review]

**Comment by JZ4012** [comment]
i take no responsibility if this breaks prod

**Review by aXises — APPROVED** [review]

**Pull request merged**